### ISSUE TITLE
@types/q: q.resolve requires parameter

### ISSUE THREAD

**Issue by craxal** [DefinitelyTyped/DefinitelyTyped#18188]
- [x] I tried using the `@types/q` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @bnemetchek @AndrewGaspar @johnnyreilly

Let's say I have the following function and compile using `typescript@2.4.1`, `q@1.5.0`, and `@types/q@1.0.2`:
```ts
function myFunction(): Promise<void> {
    return q.resolve<void>()
}
```
Using typescript@2.4.1, q@1.5.0, and @types/q@1.0.2, I get a Typescript error saying that resolve expects one argument. I have to pass `undefined` in to `q.resolve` to work around the error. I shouldn't have to pass an argument in if the return type is `void`.

A similar error can be seen when using `q.Promise<void>(...)`. The `resolve` function shouldn't need a parameter when the return type is `void`.

**Comment by MikeDimmickMnetics** [comment]
Works for me in TypeScript 2.6.2 and @types/q@1.0.6. It was fixed in merge request #18200, which merged about a week after this issue was opened.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
